mito-ai/src/utils/codeDiff.tsx

```diff
@@ -211,4 +220,42 @@ export const getCodeDiffsAndUnifiedCodeString = (originalCode: string | undefine
         unifiedCodeString,
         unifiedDiffs
     }
+}
+
+export const applyDiffStripesToCell = (
+    notebookTracker: INotebookTracker,
+    cellId: string,
+    unifiedDiffs: UnifiedDiffLine[] | null,
+    codeDiffStripesCompartments: Map<string, Compartment>
+): void => {
+    const extension = unifiedDiffs ? codeDiffStripesExtension({ unifiedDiffLines: unifiedDiffs }) : [];
+    applyCellEditorExtension(notebookTracker, cellId, extension, codeDiffStripesCompartments);
+}
+
+export const turnOffDiffsForCell = (
+    notebookTracker: INotebookTracker,
+    cellId: string,
+    codeDiffStripesCompartments: Map<string, Compartment>
+): void => {
+    applyDiffStripesToCell(notebookTracker, cellId, null, codeDiffStripesCompartments);
+}
+
+export const shouldShowDiffToolbarButtons = (
+    notebookTracker: INotebookTracker,
+    cellStateBeforeDiff: ICellStateBeforeDiff | undefined,
+    cellStatesBeforeDiff: Map<string, string>
+): boolean => {
+    try {
+        const activeCellId = notebookTracker.activeCell?.model.id;
+        if (!activeCellId) return false;
+        
+        // Check both single-cell mode and multi-cell mode
```

Comment:
Can you expand on this comment to explain why we need to be doing this? This feels like something we're going to look back on in two months and be pretty confused.